cpp/tests/io/parquet_test.cpp

```diff
@@ -5113,4 +5120,48 @@ TEST_P(ParquetSizedTest, DictionaryTest)
   EXPECT_EQ(nbits, GetParam());
 }
 
+TYPED_TEST(ParquetReaderSourceTest, BufferSourceTypes)
```

Comment:
should we cover const versions of the byte types here as well?

Reply:
I was thinking about this as well this morning. I worry about adding a pile more tests, so I'll probably add it inside to test non-const and const of a type in a single function.